src/http/oauth2.ts

```diff
@@ -1,9 +1,262 @@
 import { FetchMiddleware } from './fetcher';
-import { OAuth2Options, OAuth2, OAuth2Token } from 'fetch-mw-oauth2';
+import { OAuth2Client, OAuth2Fetch } from '@badgateway/oauth2-client';
 
-export default (oauth2Options: OAuth2Options & Partial<OAuth2Token>, token?: OAuth2Token): FetchMiddleware => {
```

Comment:
Are these types fully compatible? If not, wouldn't it be best to just fully remove this now instead of multiple breaking change releases?

Reply:
It is slightly different... maybe it _is_ better to rip off the bandaid. I think my concern was that I don't really _love_ the new API and im a bit worried it's too complicated. So in my head this was a smoother transition until i figure out the final API. but IDK. Needs some thought i guess.